clientv3/integration/watch_test.go

```diff
@@ -297,3 +297,22 @@ func putAndWatch(t *testing.T, wctx *watchctx, key, val string) {
 		}
 	}
 }
+
+func TestWatchInvalidFutureRevision(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	w := clientv3.NewWatcher(clus.RandClient())
+	defer w.Close()
+
+	rch := w.Watch(context.Background(), "foo", clientv3.WithRev(100))
+	wresp, ok := <-rch
+	if ok != false {
+		t.Fatalf("expected wresp 'closed'(ok false), but got ok %v", ok)
```

Comment:
`retc` can be left hanging; it's used internally to get the channel (`ret`) to the client

The channel should get one `WatchResponse` (the cancelled one), then close. The test needs the old bit where it reads off the message plus the new bit with the close check.